taichi/runtime/program_impls/llvm/llvm_program.cpp

```diff
@@ -120,17 +119,11 @@ void LlvmProgramImpl::materialize_snode_tree(SNodeTree *tree,
 
 std::unique_ptr<AotModuleBuilder> LlvmProgramImpl::make_aot_module_builder(
     const DeviceCapabilityConfig &caps) {
-  if (config->arch == Arch::x64 || config->arch == Arch::arm64) {
-    return std::make_unique<cpu::AotModuleBuilderImpl>(
-        *config, this, *runtime_exec_->get_llvm_context());
-  }
-
-#if defined(TI_WITH_CUDA)
-  if (config->arch == Arch::cuda) {
-    return std::make_unique<cuda::AotModuleBuilderImpl>(
-        *config, this, *runtime_exec_->get_llvm_context());
+  if (config->arch == Arch::x64 || config->arch == Arch::arm64 ||
```

Comment:
OOC: why not `arch_uses_llvm`? Is it because dx12 backend? :P

Reply:
Yes, the `dx12::AotModuleBuilder` is not inherited from 'LlvmAotModuleBuilder'.